### PR TITLE
Feature/user profile page/story 19 and 20

### PR DESCRIPTION
**Pull request by mintona**
**Functionality:**
- This branch encompasses the ability for a user to view their information on their profile page (user show page) (story 19) and to edit that information (story 20).
- A change was made to the user model to only require a password on the create action. You can see this in the file. I am not sure if this will need to be changed later, so pointing it out here. **UPDATE:** This change was revoked, as the password field is required on the edit form. 
**UPDATE:** The edit form now has the password box and the user must be authenticated before `user.update(user_params)` will run. 

**Testing:**
- All tests are passing in rspec(100%) and working in local development.

**Related Issues:**
#16 
#19 
- Notes: I did make a partial thinking we could use that for both the new and edit views for users, but we might not be able to. 
- As of now, the edit form does NOT include a field for password at all. Asked Meg and Mike about it and waiting for clarification about if that should be shown on the page or not. 
- **UPDATE:** The edit form DOES need a password field, this has been accounted for in the most recent commits.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,8 +20,39 @@ def create
     end
   end
 
+  def edit
+    @user = current_user
+  end
+
+  def update
```
Comment (DavidBarriga-Gomez) [inline app/controllers/users_controller.rb:32]: we can look into making this a private action later

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<%= form_tag path, method: method do %>
```
Comment (DavidBarriga-Gomez) [inline app/views/users/_form.html.erb:1]: nice use of a partial

**Review by DavidBarriga-Gomez — COMMENTED** [review]
Nice work

**Review by hale4029 — APPROVED** [review]
Looks legit to me, thank you.

**Review by Capleugh — APPROVED** [review]

**Pull request merged**